ported/libibverbs/rc_pingpong.c

```diff
@@ -441,9 +448,15 @@ int main(int argc, char *argv[])
 
 		switch (c) {
 		case 'S':
-			size = strtol(optarg, NULL, 0);
+			errno  = 0;
+			temp = strtol(optarg, &ptr, 10);
+                        if (ptr == optarg || *ptr != '\0' || errno == ERANGE || temp < INT_MIN || temp > INT_MAX) {
+                                fprintf(stderr, "Cannot convert from string to int\n");
+				rc = 1;
```

Comment:
@patrickmacarthur didn't know that, it makes sense though (the previous error could have been ERANGE).

If my reading of the man page is correct: if the value returned is LONG_* and errno == ERANGE, it is an error. If the value returned is LONG_* and errno != ERANGE, then it is just a really big or really small value.

This is one of those man pages that really needs an example section.